packages/zoe/src/contracts/barterExchange.js

```diff
@@ -1,136 +1,141 @@
 // @ts-check
 
 import harden from '@agoric/harden';
+import makeStore from '@agoric/store';
 import { makeZoeHelpers, defaultAcceptanceMsg } from '../contractSupport';
 
-/** @typedef {import('../zoe').ContractFacet} ContractFacet */
-
 /**
+ * This barter exchange accepts offers to trade arbitrary goods for other
+ * things. It doesn't require registration of Issuers. If two offers satisfy
+ * each other, the amounts specified in each side's `want` clause are exchanged.
+ *
  * The Barter Exchange only accepts offers that look like
  * { give: { In: amount }, want: { Out: amount} }
- *
  * The want and give amounts are both treated as minimums. Each successful
```

Comment:
I'm a bit confused as to how "want" works as a minimum. My common sense reading of that is you'd somehow be willing to accept more "want" if you could get it for your maximum "give" price? For example, if you had a give 10 and want 20, but there was an counteroffer already out there for want 10 and give 25, you'd take it, getting more than the 20 specified? Probably worth specifying what's meant by both minimums.

Reply:
Contracts can differ on how they allocate the surplus (the 5 more you mentioned). I'm leaning towards letting the original offer keep any surplus but another interesting variation would be if the contract keeps it and gives it to the instantiator.